pulsar-broker/src/main/java/org/apache/pulsar/broker/admin/impl/PersistentTopicsBase.java

```diff
@@ -2089,20 +2089,23 @@ public void readEntryComplete(Entry entry, Object ctx) {
     }
 
     protected Response internalPeekNthMessage(String subName, int messagePosition, boolean authoritative) {
-        verifyReadOperation(authoritative);
+        verifyReadOperation();
+
+        TopicName partitionedTopicName = topicName;
         // If the topic name is a partition name, no need to get partition topic metadata again
         if (!topicName.isPartitioned() && getPartitionedTopicMetadata(topicName, authoritative, false).partitions > 0) {
-            throw new RestException(Status.METHOD_NOT_ALLOWED, "Peek messages on a partitioned topic is not allowed");
+            partitionedTopicName = topicName.getPartition(0);
```

Comment:
@hangc0276 I think it is true that we don't want to support peeking messages on a partitioned topic. But we want to support peeking messages on a partition of a partitioned topic.

E.g. if topic `test-topic` is a partitioned topic, it should fail if a user attempts to peek messages on topic `test-topic`. But it should succeed if a users attempts to peek messages from its partitions like `test-topic-partition-0`.

Reply:
Yes, if we support peeking messages on a partitioned topic, it may cause message inconsistent. I my current implementation, i just peek the partition 0 messages.
In this case,  issue #8174 can be closed.

Reply:
@sijie Yes, I Can't reproduce the case of #8174 described by @tuteng . Maybe the Pulsar manager peeking message is actually called Pulsar admin API described by me.

Reply:
I see. If that's the case, we need to fix the problem in Pulsar Manager. Pulsar Manager should use the partition name to peek messages.